build_cmake/CMakeLists.txt

```diff
@@ -2,15 +2,40 @@ cmake_minimum_required(VERSION 2.8)
 project(CodeGen)
 
 set(CMAKE_MODULE_PATH
-	${CodeGen_SOURCE_DIR}/Modules
+	${CMAKE_CURRENT_SOURCE_DIR}/../../Dependencies/cmake-modules
 	${CMAKE_MODULE_PATH}
 )
 
+if(NOT CMAKE_BUILD_TYPE)
+	set(CMAKE_BUILD_TYPE Release CACHE STRING
+		"Choose the type of build, options are: None Debug Release"
+		FORCE)
+endif()
+
+if(NOT MSVC)
+	if(CMAKE_BUILD_TYPE STREQUAL "Release")
+		add_definitions(-DNDEBUG)
+	else()
+		add_definitions(-D_DEBUG)
+	endif()
+endif()
+
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 if (NOT TARGET Framework)
-	add_subdirectory(../../Framework/build_unix ${CodeGen_SOURCE_DIR}/../../Framework/build_unix/build)
+	add_subdirectory(
+		${CMAKE_CURRENT_SOURCE_DIR}/../../Framework/build_cmake/Framework
+		${CMAKE_CURRENT_SOURCE_DIR}/../../Framework/build_cmake/Framework/build_${CMAKE_SYSTEM_NAME}
```

Comment:
great this seems to work, the trade of is, we just lose the advantage of having each repo build being self contained, but its not much but a small convenience.

my eyes are shutting of, I'll have a commit out by 2morrow night... or if paradoxically I can't sleep :/